roles/kvirt_vm/tasks/delete-vm.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: "Delete VM {{ kvirt_vm_name }}"
+  community.kubernetes.k8s:
+    kind: VirtualMachine
+    api_version: "{{ kvirt_vm_api_version }}"
+    name: "{{ kvirt_vm_name }}"
+    namespace: "{{ kvirt_vm_namespace }}"
+    state: absent
+
+- name: Wait for VirtualMachine to be deleted
+  community.kubernetes.k8s_info:
+    api: "{{ kvirt_vm_api_version }}"
+    kind: VirtualMachine
+    name: "{{ kvirt_vm_name }}"
+    namespace: "{{ kvirt_vm_namespace }}"
+  register: kvirt_vm_info
+  until:
+    - kvirt_vm_info.resources is defined
+    - kvirt_vm_info.resources | length == 0
+  retries: 60
+  delay: 5
+...
```

Comment:
No need to check if you wait

Reply:
Not all the resources have the wait implemented, that works common kinds. It would work with VirtualMachines along with wait_condition if there were conditions that we can use, but that is not the case:
```
status:
  conditions:
  - lastProbeTime: "2024-01-31T22:43:14Z"
    lastTransitionTime: "2024-01-31T22:43:14Z"
    message: VMI does not exist
    reason: VMINotExists
    status: "False"
    type: Ready
```

I tried using only wait but the module exists immediately.

Reply:
Ah you're right, thanks for checking!

Would be worth doing the wait_condition in the same task or you rather do it as it is in a separate task?